app/lib/screens/app_onboarding/preview.dart

```diff
@@ -66,15 +66,17 @@ class Preview {
     // delete study subscription and progress
     if (containsQueryPair('cmd', 'reset')) {
       if (selectedStudyObjectId != null) {
-        final StudySubject subject = await SupabaseQuery.getById<StudySubject>(
-          selectedStudyObjectId!,
-          selectedColumns: [
-            '*',
-            'study!study_subject_studyId_fkey(*)',
-            'subject_progress(*)',
-          ],
-        );
-        subject.delete();
+        final List<StudySubject> subjects =
+            await SupabaseQuery.getAll<StudySubject>();
+
+        final List<StudySubject> subjectsToDelete = subjects
+            .where((subject) =>
+                subject.userId == Supabase.instance.client.auth.currentUser!.id,)
+            .toList();
+
+        subjectsToDelete.forEach((subject) async {
+          await subject.delete();
+        });
```

Comment:
Could you revert the changes here so this PR is independent of  https://github.com/hpi-studyu/studyu/pull/699